controller/src/main/java/org/jboss/as/controller/AbstractAttributeDefinitionBuilder.java

```diff
@@ -907,6 +909,9 @@ public String getAttributeGroup() {
         if (toCopy == null) {
             return null;
         }
+        if (toCopy.length == 0){
+            return ZERO_CONSTRAINTS;
```

Comment:
In the end this value goes into AttributeDefinition.wrapConstraints. If you tweak that to return Collections.emptyList() for a zero-sized array, that will save a bit more.